### ISSUE TITLE
Do not change default port wireguard in wg0.conf and peer#.conf

### ISSUE THREAD

**Issue by ddd015** [linuxserver/docker-wireguard#176]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior

If I change option SERVERPORT (docker-compose file) then it should change in files wg0.conf peer#.conf
## Current Behavior

Does not apply option SERVERPORT in docker-compose file when generating files wg0.conf peer#.conf
## Steps to Reproduce

1.
2.
3.
4.

## Environment
**OS:**
**CPU architecture:** x86_64/arm32/arm64
**How docker service was installed:**

**Comment by ddd015** [comment]
Do not change default port wireguard

**Comment by aptalca** [comment]
Please stop opening new issues for the same thing.

`SERVERPORT` is NOT `listen port`. It's all explained in the readme and is working as expected.

If you are going to open a new issue for legitimate bug, please fill out the issue template as requested.

**Comment by Friday13th87** [comment]
> Please stop opening new issues for the same thing.
> 
> `SERVERPORT` is NOT `listen port`. It's all explained in the readme and is working as expected.
> 
> If you are going to open a new issue for legitimate bug, please fill out the issue template as requested.

There is absolutely nothing explained in the readme, the only thing written there is
"External port for docker host. Used in server mode."

This question is coming up pretty often and the only thing you are always doing is responding in a not-helpful and ignorant way instead of clearing up things only once.
You could just clearing it on the readme itself, that people can use any external server port but need to take the default listening port anyway but you are not even willing to clear it up for the people who dont know.

**Comment by aptalca** [comment]
from the readme:

| Parameter | Function |
| :----: | --- |
| `-e SERVERURL=wireguard.domain.com` | External IP or domain name for docker host. Used in server mode. If set to `auto`, the container will try to determine and set the external IP automatically |
| `-e SERVERPORT=51820` | External port for docker host. Used in server mode. |

They both say `External ____`. One's for url, the other's for port. They are listed together, one's right above the other. Why do you assume the external port is also going to change the internal port, but you don't assume the external ip is going to change the internal binding ip?
I believe the readme is clear.

> This question is coming up pretty often

Because lots of people tend not to read

**Comment by ddshd** [comment]
So is there no way to change the listen port?

**Comment by thespad** [comment]
There is no way *as part of the automated setup* to change the *client* listen port or the *internal* server listen port, no.
